core/liquidity/v2/snapshot_v1.go

```diff
@@ -0,0 +1,213 @@
+	stopped bool
+	keys    []string
+}
+
+func (e *snapshotV1) Namespace() types.SnapshotNamespace {
+	return types.LiquiditySnapshot
+}
+
+func (e *snapshotV1) Keys() []string {
+	if len(e.keys) <= 0 {
+		e.keys = []string{
+			(&types.PayloadLiquidityParameters{
+				Parameters: &snapshotpb.LiquidityParameters{
+					MarketId: e.market,
+				},
+			}).Key(),
+			(&types.PayloadLiquidityPendingProvisions{
+				PendingProvisions: &snapshotpb.LiquidityPendingProvisions{
+					MarketId: e.market,
+				},
+			}).Key(),
+			(&types.PayloadLiquidityProvisions{
+				Provisions: &snapshotpb.LiquidityProvisions{
+					MarketId: e.market,
+				},
+			}).Key(),
+			(&types.PayloadLiquiditySupplied{
+				LiquiditySupplied: &snapshotpb.LiquiditySupplied{
+					MarketId: e.market,
+				},
+			}).Key(),
+			(&types.PayloadLiquidityScores{
+				LiquidityScores: &snapshotpb.LiquidityScores{
+					MarketId: e.market,
+				},
+			}).Key(),
+		}
+	}
+
+	return e.keys
+}
+
+func (e *snapshotV1) GetState(k string) ([]byte, []types.StateProvider, error) {
+	var (
+		keys  = e.Keys()
+		state = &snapshotpb.Payload{}
+	)
+
+	switch k {
+	case keys[0]:
+		state.Data = &snapshotpb.Payload_LiquidityParameters{
+			LiquidityParameters: &snapshotpb.LiquidityParameters{
+				MarketId: e.market,
+			},
+		}
+	case keys[1]:
+		state.Data = &snapshotpb.Payload_LiquidityPendingProvisions{
+			LiquidityPendingProvisions: &snapshotpb.LiquidityPendingProvisions{
+				MarketId: e.market,
+			},
+		}
+	case keys[2]:
+		state.Data = &snapshotpb.Payload_LiquidityProvisions{
+			LiquidityProvisions: &snapshotpb.LiquidityProvisions{
+				MarketId: e.market,
+			},
+		}
+	case keys[3]:
+		state.Data = &snapshotpb.Payload_LiquiditySupplied{
+			LiquiditySupplied: &snapshotpb.LiquiditySupplied{
+				MarketId: e.market,
+			},
+		}
+	case keys[4]:
+		state.Data = &snapshotpb.Payload_LiquidityScores{
+			LiquidityScores: &snapshotpb.LiquidityScores{
+				MarketId: e.market,
+			},
+		}
+	}
+
+	buf, err := proto.Marshal(state)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return buf, nil, nil
+}
+
+func (e *snapshotV1) LoadState(ctx context.Context, p *types.Payload) ([]types.StateProvider, error) {
+	if e.Namespace() != p.Data.Namespace() {
+		return nil, types.ErrInvalidSnapshotNamespace
+	}
+
+	switch pl := p.Data.(type) {
+	case *types.PayloadLiquidityProvisions:
+		if err := e.loadProvisions(ctx, pl.Provisions.GetLiquidityProvisions()); err != nil {
+			return nil, err
+		}
+
+		e.loadPerformances(pl.Provisions.GetLiquidityProvisions())
+		return nil, nil
+	case *types.PayloadLiquidityScores:
+		return nil, e.loadScores(pl.LiquidityScores)
+	case *types.PayloadLiquiditySupplied:
+		return nil, e.loadSupplied(pl.LiquiditySupplied)
+	case *types.PayloadLiquidityPendingProvisions:
+		return nil, nil
+	case *types.PayloadLiquidityParameters:
+		return nil, nil
+
+	default:
+		return nil, types.ErrUnknownSnapshotType
+	}
+}
+
+func (e *snapshotV1) Stopped() bool {
+	return e.stopped
```

Comment:
Wonder whether this should always return `true` in the V1 case so that it'll get removed at the next snapshot? As far as I can see currently it'll still hang around as a provider and still be providing state etc etc. and I don't know if that will cause problems at any next snapshot restore that happens.

Reply:
Yea I wondered too. And it broke at first when I tried to do it. So instead made it return true until the market is removed, but with an empty state.